appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/service/FaultToleranceServiceImpl.java

```diff
@@ -200,9 +205,17 @@ public FaultToleranceConfig getConfig(InvocationContext context, Stereotypes ste
 
     @Override
     public FaultToleranceMetrics getMetrics(InvocationContext context) {
+
         FaultToleranceApplicationState appState = getApplicationState(getApplicationContext(context));
-        return appState.getMetrics().updateAndGet(
-                metrics -> metrics != null ? metrics : new BindableFaultToleranceMetrics()).bindTo(context);
+        return appState.getMetrics(this::getApplicationMetricRegistry, FaultToleranceUtils.getCanonicalMethodName(context));
+    }
+
+    private MetricRegistry getApplicationMetricRegistry() {
```

Comment:
How about using an optional here?

Reply:
PR is still in development. Let me get back when I feel ready.

Reply:
sure

Reply:
@svendiedrichsen I had a look now. The return value is just used internally. I didn't feel that `Optional` would have improved the code. This might be because I am not a big fan personally or because `Optional` isn't use in our codebase generally. I don't feel starting here is the way to go. Not even sure I ever want to start using it but if this should be something broadly applied to the codebase. To go from having to expect `null` basically everywhere to _there should not be null references_ and if that means you need an `Optional` then use one.